### PR TITLE
add check for canonical point encodings where needed, and tests

### PR DESCRIPTION
**Pull request by conradoplg**
While adding support for Ed448 I noticed that [curve25519-dalek](https://github.com/dalek-cryptography/curve25519-dalek/issues/380) and Ed448-Goldilocks didn't check for non-canonical point encodings (e.g. with Y coordinate >= p). Turns out this isn't needed for Ed25519 since we check for non-prime order points and that also blocks non-canonical points, but it was needed for Ed448. I also added tests for all ciphersuites.

### CODE REVIEW

File: frost-ed448/src/lib.rs [hunk 1]
```diff
@@ -93,13 +93,20 @@ impl Group for Ed448Group {
...
+        let compressed = CompressedEdwardsY(*buf);
+        match compressed.decompress() {
             Some(point) => {
                 if point == Self::identity() {
                     Err(GroupError::InvalidIdentityElement)
                 } else if point.is_torsion_free() {
-                    Ok(point)
+                    // decompress() does not check for canonicality, so we
+                    // check by recompressing and comparing
+                    if point.compress().0 != compressed.0 {
```
Comment (dconnolly) [inline frost-ed448/src/lib.rs:105]: 👍

**Review by dconnolly — DISMISSED** [review]
Great!

**Comment by dconnolly** [comment]
Oh I may have done an incomplete merge conflict resolve

**Comment by dconnolly** [comment]
@mergify rebase

**Comment by conradoplg** [comment]
@dconnolly fixed it!

**Comment by conradoplg** [comment]
Now it's really fixed!

**Review by mpguerra — APPROVED** [review]
Re-approving to get this merged.

**Pull request merged**